drivers/dac/dac_dacx0508.c

```diff
@@ -0,0 +1,463 @@
+			 "below the required minimum analog threshold");
+		return -EIO;
+	}
+
+	data->configured[channel_cfg->channel_id] = true;
+
+	return 0;
+}
+
+static int dacx0508_write_value(struct device *dev, uint8_t channel,
+				uint32_t value)
+{
+	const struct dacx0508_config *config = dev->config_info;
+	struct dacx0508_data *data = dev->driver_data;
+	uint8_t regval[2];
+	int ret;
+
+	if (!data->configured[channel]) {
+		LOG_ERR("channel not initialized");
+		return -EINVAL;
+	}
+
+	if (channel > DACX0508_MAX_CHANNEL - 1) {
+		LOG_ERR("unsupported channel %d", channel);
+		return -ENOTSUP;
+	}
+
+	if (value >= (1 << config->resolution)) {
+		LOG_ERR("value %d out of range", value);
+		return -EINVAL;
+	}
+
+	regval[0] = value >> 8;
+	regval[1] = value & 0xFF;
+
+	ret = dacx0508_write(dev, DACX0508_REG_DAC0 + channel, regval);
+	if (ret) {
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int dacx0508_soft_reset(struct device *dev)
+{
+	uint8_t regval[2] = {0, DACX0508_MASK_TRIGGER_SOFT_RESET};
+	int ret;
+
+	ret = dacx0508_write(dev, DACX0508_REG_TRIGGER, regval);
+	if (ret) {
+		return -EIO;
+	}
+	k_usleep(DACX0508_POR_DELAY);
+
+	return 0;
+}
+
+static int dacx0508_device_id_check(struct device *dev)
+{
+	const struct dacx0508_config *config = dev->config_info;
+	uint8_t regval[2] = {0, };
+	uint16_t dev_id;
+	int ret;
+
+	ret = dacx0508_read(dev, DACX0508_REG_DEVICE_ID, regval);
+	if (ret) {
+		LOG_ERR("Unable to read Device ID");
+		return -EIO;
+	}
+	dev_id = (regval[0] << 8) | regval[1];
+
+	switch (config->resolution) {
+	case 12:
+		if ((dev_id & DACX0508_MASK_DEVICE_ID_12BIT) !=
+					DACX0508_MASK_DEVICE_ID_12BIT) {
+			LOG_ERR("not match chip resolution");
+			return -EINVAL;
+		}
+		break;
+	case 14:
+		if ((dev_id & DACX0508_MASK_DEVICE_ID_14BIT) !=
+					DACX0508_MASK_DEVICE_ID_14BIT) {
+			LOG_ERR("not match chip resolution");
+			return -EINVAL;
+		}
+		break;
+	case 16:
+		if ((dev_id & DACX0508_MASK_DEVICE_ID_16BIT) != 0) {
+			LOG_ERR("not match chip resolution");
+			return -EINVAL;
+		}
+		break;
+	default:
+		LOG_ERR("unsupported resolution %d", config->resolution);
+		return -ENOTSUP;
+	}
+
+
+	if ((dev_id & DACX0508_MASK_DEVICE_ID_8CH) !=
+				DACX0508_MASK_DEVICE_ID_8CH) {
+		LOG_ERR("Support channles mismatch");
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int dacx0508_init(struct device *dev)
+{
+	const struct dacx0508_config *config = dev->config_info;
+	struct dacx0508_data *data = dev->driver_data;
+	int ret;
+
+	data->spi_dev = device_get_binding(config->spi_dev_name);
+	if (!data->spi_dev) {
+		LOG_ERR("Cannot get pointer to %s device",
+			    config->spi_dev_name);
+		return -EINVAL;
+	}
+
+	data->spi_cs.gpio_dev = device_get_binding(config->spi_cs_dev_name);
+	if (!data->spi_cs.gpio_dev) {
+		LOG_ERR("Cannot get pointer to %s device",
+		       config->spi_cs_dev_name);
+		return -EINVAL;
+	}
+	data->spi_cs.gpio_pin = config->spi_cs_pin;
+	data->spi_cfg = config->spi_cfg;
+	data->spi_cfg.cs = &data->spi_cs;
```

Comment:
This needs to take into account, that `cs-gpios` is optional. See e.g the `dac_lmp90xxx.c` driver for an example on how this can be done:
https://github.com/zephyrproject-rtos/zephyr/blob/9c39a930148d3ab843c11e071c2741251cc75764/drivers/adc/adc_lmp90xxx.c#L939